apps/wpcom-block-editor/src/wpcom/features/live-preview/upgrade-button.tsx

```diff
@@ -1,51 +1,51 @@
 import { useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { useEffect } from 'react';
-import tracksRecordEvent from '../../tracking/track-record-event';
-import { getUnlock } from '../utils';
-import { usePreviewingTheme } from './use-previewing-theme';
+import { FC, useEffect } from 'react';
+import tracksRecordEvent from '../tracking/track-record-event';
+import { usePreviewingTheme } from './hooks/use-previewing-theme';
+import { getUnlock } from './utils';
+
+import './upgrade-button.scss';
 
 const SAVE_HUB_SAVE_BUTTON_SELECTOR = '.edit-site-save-hub__button';
 const HEADER_SAVE_BUTTON_SELECTOR = '.edit-site-save-button__button';
 
 const unlock = getUnlock();
 
-/**
- * This overrides the `SaveButton` behavior by adding a listener and changing the copy.
- * Our objective is to open a custom modal ('ThemeUpgradeModal') instead of proceeding with the default behavior.
- * For more context, see the discussion on adding an official customization method: https://github.com/WordPress/gutenberg/pull/56807.
- */
-export const useOverrideSaveButton = ( {
-	setIsThemeUpgradeModalOpen,
-	previewingTheme,
-}: {
-	setIsThemeUpgradeModalOpen: ( isThemeUpgradeModalOpen: boolean ) => void;
+export const LivePreviewUpgradeButton: FC< {
 	previewingTheme: ReturnType< typeof usePreviewingTheme >;
-} ) => {
+	upgradePlan: () => void;
+} > = ( { previewingTheme, upgradePlan } ) => {
 	const canvasMode = useSelect(
 		( select ) =>
 			unlock && select( 'core/edit-site' ) && unlock( select( 'core/edit-site' ) ).getCanvasMode(),
 		[]
 	);
 
+	/**
+	 * This overrides the `SaveButton` behavior by adding a listener and changing the copy.
+	 * Our objective is to open the Plans page instead of proceeding with the default behavior.
+	 * For more context, see the discussion on adding an official customization method: https://github.com/WordPress/gutenberg/pull/56807.
+	 */
 	useEffect( () => {
 		const saveButtonClickHandler: EventListener = ( e ) => {
 			e.preventDefault();
 			e.stopPropagation();
-			setIsThemeUpgradeModalOpen( true );
 			tracksRecordEvent( 'calypso_block_theme_live_preview_upgrade_modal_open', {
```

Comment:
I've noticed that some of our tracking event names no longer accurately reflect their functionality. 
Should we change the name of track events, or not? For example, this `calypso_block_theme_live_preview_upgrade_modal_open` suggests it triggers when a modal opens, but this action no longer occurs. Similarly, we have the `calypso_block_theme_live_preview_upgrade_modal_upgrade` event inside the `upgradePlan` function, which is no longer triggered by the button on the modal. 

The current funnel: https://mc.a8c.com/tracks/funnel/view.php?52154/2024-01-10/to/2024-04-18&frame-nonce=6978a9dafa

Reply:
Yeah, this is intentional. I wrote the following in the Note section:

> I am still using the same calypso_block_theme_live_preview_upgrade_modal_open track event even though it's now opening a new tab (instead of a modal). This is so that we can analyze track trend/funnel with the same event.

This was done so that we can see the before/after graph in the same funnel with the same Tracks event. Will this be useful or simplify the analysis?

We can rename the event later after we have the analysis (if we want to). What do you think?

Reply:
Sorry for missing the note! 

>We can rename the event later after we have the analysis (if we want to)

Keeping the events as is for now makes sense to me. Thanks for the clarification!